### PR TITLE
Milo/duplicate plan button

### LINKED ISSUE #690: Duplicate Plan
Users may want to duplicate their plans so that they can easily build off their current plans but with differing classes/majors. Implement a way to duplicate a plan over to a new plan.

This ticket involves no design work, so just implement the functionality and place it into a button in the edit plan modal.
If you'd like (milo), you can do some thinking as to how the design would work. 
Personally, I think a good flow would be when you open the modal to create a new plan, it gives you the option to "base it off of a preexisting plan" which would essentially just be a dropdown of your previous plans.

In the next sprint, there will be a follow up ticket for implementing the design for this flow, so just make sure the technical implementation is kept as simple/modular as possible, i.e. not as many dependencies etc.

### LINKED ISSUE #692: Implement new duplicate plan designs
Technical implementation for duplicate plan has already been done in #690. All we need to do now is implement the designs created by our designers. Contact Angela/Elaine to get the figma designs and implement the proper workflow.

### PR DESCRIPTION
**Pull request by yijen-sun**
# Description
Added duplicate plan button that creates a copy of the current plan as a new plan.
Disabled the Add Plan button and the Duplicate Plan button on guest mode when the maximum number of plans (5) has been reached. This is because the local storage that guest data is stored on is limited and will cause issues lol

https://github.com/sandboxnu/graduatenu/assets/94196677/936db3f7-a3bf-4d8a-af8e-870569c24ac6

Closes #690
Closes #692 

## Type of change

Please tick the boxes that best match your changes.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`
- [ ] This has migration changes and requires a run of `yarn dev:migration:run`

# How Has This Been Tested?

Please describe how you tested this PR (both manually and with tests) Provide instructions so we can reproduce.

Manually tested 

# Checklist:

- [X] I have run the production builds in docker for the frontend/backend and ensure things run fine. Check README of repo on how to run if not sure.
- [X] I have performed a self-review of my own code
- [X] I have commented my code where needed
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I've run the end to end tests
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AngelaZQ1** [comment]
404k additions 💀

File: 46mb.json
Comment (KobeZ123) [inline 46mb.json:1]: should this file be deleted? i'm assuming it's the local storage data.

File: ls.json
Comment (KobeZ123) [inline ls.json:1]: are we keeping this test plan for reference?

File: packages/frontend/components/Plan/DuplicatePlanButton.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  setSelectedPlanId,
+}) => {
+  const router = useRouter();
+  const { isGuest } = useContext(IsGuestContext);
+  const [buttonLoading, setButtonLoading] = useState(false);
+  const { student } = useStudentWithPlans();
+
+  const duplicatePlan = async () => {
+    if (!student) return;
+    // TODO: figure out when to do with this -- debounce stuff
```
Comment (KobeZ123) [inline packages/frontend/components/Plan/DuplicatePlanButton.tsx:38]: is this TODO still relevant? i think adding the plan limit was our solution, right?

File: packages/frontend/pages/home.tsx [hunk 2]
```diff
@@ -34,10 +34,12 @@ import {
   Sidebar,
   TransferCourses,
 } from "../components";
+import { DuplicatePlanButton } from "../components/Plan/DuplicatePlanButton";
 import { fetchStudentAndPrepareForDnd, useStudentWithPlans } from "../hooks";
 import {
-  cleanDndIdsFromPlan,
+  // createEmptySchedule,
```
Comment (KobeZ123) [inline packages/frontend/pages/home.tsx:40]: we can remove this

**Review by KobeZ123 — CHANGES_REQUESTED** [review]

**Review by KobeZ123 — APPROVED** [review]
Nice work!

**Pull request merged**